src/nupic/encoders/Base.hpp

```diff
@@ -0,0 +1,57 @@
+/* ---------------------------------------------------------------------
+ * Numenta Platform for Intelligent Computing (NuPIC)
+ * Copyright (C) 2016, Numenta, Inc.  Unless you have an agreement
+ * with Numenta, Inc., for a separate license for this software code, the
+ * following terms and conditions apply:
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Affero Public License version 3 as
+ * published by the Free Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
+ * See the GNU Affero Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero Public License
+ * along with this program.  If not, see http://www.gnu.org/licenses.
+ *
+ * http://numenta.org/licenses/
+ * ---------------------------------------------------------------------
+ */
+
+/** @file
+ * Defines the abstract Encoder base class
+ */
+
+#ifndef NTA_ENCODERS_BASE
+#define NTA_ENCODERS_BASE
+
+#include <nupic/ntypes/ArrayBase.hpp>
+#include <nupic/types/Types.hpp>
+
+namespace nupic
+{
+  /** An encoder converts a value to a sparse distributed representation.
```

Comment:
this is not true, but I guess you've just copied the line from python? 
An encoder converts a "value" (real world entity) to a binary vector (typically dense, not SDR), SP does that. 
Btw, you could link the arXiv paper on Encoders by Scott and Alvin in the doc